src/vm/types/pedersen_instance_def.zig

```diff
@@ -20,5 +27,113 @@ pub const PedersenInstanceDef = struct {
     /// The upper bound on the hash inputs.
     ///
     /// If None, the upper bound is 2^element_bits.
-    _hash_limit: std.math.big.int.Managed,
+    _hash_limit: ManagedBigInt,
+    /// The memory allocator. Can be needed for the deallocation of the VM resources.
+    allocator: Allocator,
+
+    pub fn default(allocator: Allocator) !Self {
+        var limbs = try allocator.alloc(Limb, 8);
+        limbs[0] = 1;
+        limbs[1] = 0;
+        limbs[2] = 0;
+        limbs[3] = 0;
+        limbs[4] = 0;
+        limbs[5] = 0;
+        limbs[6] = 17;
+        limbs[7] = 134217728;
+
+        return .{ .allocator = allocator, .ratio = 8, ._repetitions = 4, ._element_height = 256, ._element_bits = 252, ._n_inputs = 2, ._hash_limit = .{ .allocator = allocator, .limbs = limbs, .metadata = 1 } };
+    }
+
+    pub fn init(allocator: Allocator, ratio: ?u32, _repetitions: u32) !Self {
+        var limbs = try allocator.alloc(Limb, 8);
+        limbs[0] = 1;
+        limbs[1] = 0;
+        limbs[2] = 0;
+        limbs[3] = 0;
+        limbs[4] = 0;
+        limbs[5] = 0;
+        limbs[6] = 17;
+        limbs[7] = 134217728;
+
+        const hash_limits = ManagedBigInt{ .allocator = allocator, .limbs = limbs, .metadata = 1 };
+        return .{ .ratio = ratio, ._repetitions = _repetitions, ._element_height = 256, ._element_bits = 252, ._n_inputs = 2, ._hash_limit = hash_limits, .allocator = allocator };
+    }
+
+    pub fn deinit(self: *Self) void {
+        self._hash_limit.deinit();
+    }
+
+    pub fn cellsPerBuiltin(_: *const Self) u32 {
+        return CELLS_PER_HASH;
```

Comment:
This one is declared as public constant so that we don't need a dedicated function for that in my opinion.

Reply:
Yes this might be a bit redundant

Reply:
ping @dhruvkelawala

Reply:
Should I also remove `rangeCheckPerBuiltin`?